src/engine/Union.cpp

```diff
@@ -17,16 +17,16 @@ Union::Union(QueryExecutionContext* qec, std::shared_ptr<QueryExecutionTree> t1,
   _columnOrigins.resize(variableColumns.size(), {NO_COLUMN, NO_COLUMN});
   for (auto it : variableColumns) {
     // look for the corresponding column in t1
-    auto it1 = t1->getVariableColumnMap().find(it.first);
-    if (it1 != t1->getVariableColumnMap().end()) {
+    auto it1 = t1->getVariableColumns().find(it.first);
+    if (it1 != t1->getVariableColumns().end()) {
```

Comment:
maybe also cache the result to `getVariableColumns()` but this is one of the shorter variants to this I suggested earlier.